### ISSUE TITLE
Watching hangs when a large number of file are created in a very short time.

### ISSUE THREAD

**Issue by itkovian** [itkovian/sarchive#20]
This issue occurs in the torque branch, where there is often a single directory to watch and the scheduler may create a large number of file when a job array is submitted.

To reproduce:
- start sarchive in torque mode, watching a single directory
- create a large number of files in this directory, e.g. 10K

Increasing the queue size through echo `65535 >  /proc/sys/fs/inotify/max_queued_events` does not always help. After creating 50 K files with 64K event queue size, the execution stopped at

~~~~
[2019-06-13T22:08:58.955897586+02:00][sarchive::lib][INFO] copied 11 bytes from "/tmp/torque/17354.SC" to "/tmp/archive/job.17354_SC"
[2019-06-13T22:08:58.956109575+02:00][sarchive::lib][INFO] copied 11 bytes from "/tmp/torque/17355.SC" to "/tmp/archive/job.17355_SC"
~~~~
  
Issue occurs with
- kernel 3.10.0-957.12.2.el7.ug.x86_64. (custom build based on the version number)
- glibc-2.17-260.el7_6.5.x86_64
- notify crate main branch with commit #56aac12
- running on a kvm VM with storage in a Ceph backend

**Comment by itkovian** [comment]
The problem is that there seems to be some deadlock occurring, due to the usage of DebouncedEvents, where the notify package tries to merge events on the same file into a single (higher level) event. However, with a large number of events, this seems to result in a deadlock, where both the watcher and the debouncing thread are in an `__lll_lock_wait`.

Using a RawEvents fixes this. Is it not clear if this has other repercussions, such as the need to wait a little bit before processing the event to make sure the files have been written, or if we need to wait for a different event, such as a closing event on the file.

**Comment by itkovian** [comment]
For torque, I think that watching for a CLOSE_WRITE event should be feasible. Slurm OTOH will create a directory, and recursive watching in a high-volume environment has proven not to be sufficiently reliable, so I'll need to think of another approach there.

**Comment by itkovian** [comment]
The problem with using RawEvents is that now, somehow, SIGINT and SIGTERM no longer shut down the program.

**Comment by itkovian** [comment]
DebouncedEvents are off the table for now in https://github.com/passcod/notify v. 5.0.0. 

In the `torque` branch, code has been moved to this version (commit d4e8e70).

**Comment by itkovian** [comment]
This can be considered to be fixed, by #22.
